examples/asgardeo-ai/obo_flow.py

```diff
@@ -0,0 +1,151 @@
+        print("2. Log in with your Asgardeo credentials")
+        print("3. Authorize the AI agent")
+        print("4. After authorization, you'll be redirected to the callback URL")
+        print("5. Copy the 'code' parameter from the callback URL")
+        print("6. Come back here and paste it below")
+        
+        # Wait for user to complete OAuth flow
+        print("Waiting for you to complete the OAuth flow...")
+        input("Press ENTER when you've opened the URL and are ready to continue...")
+        
+        # Get authorization code from user
+        print("Please paste the authorization code from the callback URL:")
+        print("   (Look for '?code=...' in the URL after authorization)")
+        authorization_code = input("Authorization code: ").strip()
+        
+        if not authorization_code:
+            print("No authorization code provided. Exiting...")
+            return
+        
+        # Step 3: Exchange authorization code for user token
+        print(f"Exchanging authorization code for user token...")
+        print(f"Using code: {authorization_code[:20]}...")
+        
+        try:
+            # Create fresh config and auth manager for OBO token exchange
+            config2 = AsgardeoConfig(
+                base_url="https://api.asgardeo.io/t/<org-name>",
+                client_id="<client-id>",
+                redirect_uri="<redirect-uri>",
+                client_secret="<client-secret>",
+                session=session2
+            )
+            
+            # Create new auth manager with fresh session for OBO operation
+            obo_auth_manager = AgentAuthManager(config2, agent_config)
+            
+            # Exchange the real authorization code for user tokens
+            user_token = await obo_auth_manager.get_obo_token(
+                auth_code=authorization_code,
+                scopes=user_scopes,
+                agent_token=agent_token
+            )
+            
+            print("SUCCESS! User token obtained via OBO flow!")
+            print(f"User Access Token: {user_token.access_token[:30]}...")
+            if user_token.id_token:
+                print(f"User ID Token: {user_token.id_token[:30]}...")
+            if user_token.refresh_token:
+                print(f"User Refresh Token: {user_token.refresh_token[:30]}...")
+            print(f"Expires in: {user_token.expires_in} seconds")
+            print(f"Scope: {user_token.scope}")
+
+            # Now the agent can act on behalf of the user
+            print("The AI agent can now:")
+            print("   • Access user's protected resources")
+            print("   • Make API calls on behalf of the user")
+            print("   • Maintain user context in AI operations")
+            print("   • Perform actions with user's permissions")
+
+            # Demonstrate token usage
+            print("Testing token validity...")
+            await asyncio.sleep(1)  # Simulate some work
```

Comment:
shall we add the token validation logic here? hope the SDK provides a method for that.